### PR TITLE
Add some new TD payload test cases

### PR DESCRIPTION
**Pull request by liuw1**
1. Add test cases for tdcall tdreport and tdvmcal io read and write. 
2. Correct the build command in doc/test_with_td_payload.md.
3. Do small changes in testtdinfo.rs.

### CODE REVIEW

File: doc/test_with_td_payload.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ $ cd ..
 ### Generate final.bin
 ```
 $ cargo xbuild -p td-shim --target x86_64-unknown-uefi --release --features=main,tdx --no-default-features
-$ cargo run -p td-shim-tools --bin td-shim-ld -- target/x86_64-unknown-uefi/release/ResetVector.bin target/x86_64-unknown-uefi/release/td-shim.efi target/x86_64-unknown-uefi/release/test-td-payload.efi -o target/x86_64-unknown-uefi/release/final-pe.bin
+$ cargo run -p td-shim-tools --bin td-shim-ld --no-default-features --features="linker" -- target/x86_64-unknown-uefi/release/ResetVector.bin target/x86_64-unknown-uefi/release/td-shim.efi target/x86_64-unknown-uefi/release/test-td-payload.efi -o target/x86_64-unknown-uefi/release/final-pe.bin
```
Comment (jiangliu) [inline doc/test_with_td_payload.md:37]: Why do we need "--no-default-features" here?

Comment (liuw1) [reply doc/test_with_td_payload.md:37]: It is specific for QEMU boot, I will remove it.

File: tests/test-td-payload/src/main.rs [hunk 2]
```diff
@@ -20,7 +23,10 @@ use td_layout::memslice;
 use tdx_tdcall::tdx;
 
 use crate::lib::{TestResult, TestSuite};
+use crate::testiorw32::Tdiorw32;
+use crate::testiorw8::Tdiorw8;
 use crate::testtdinfo::Tdinfo;
+use crate::testtdreport::Tdreport;
 use alloc::boxed::Box;
```
[the remark this answers is not in the document]
Comment (liuw1) [reply tests/test-td-payload/src/main.rs:30]: Thanks, I will update it.

File: tests/test-td-payload/src/testiorw32.rs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    fn setup(&mut self) {
+        self.result = TestResult::Fail;
+    }
+
+    /**
+     * run the test case
+     * io read Century of RTC
+     */
+    fn run(&mut self) {
+        let read1 = tdx::tdvmcall_io_read_32(0x71);
```
Comment (jyao1) [inline tests/test-td-payload/src/testiorw32.rs:41]: you need write 0x70 at first.

Comment (liuw1) [reply tests/test-td-payload/src/testiorw32.rs:43]: Updated, Thanks!

**Review by jyao1 — APPROVED** [review]

**Pull request merged**